util/timer/src/timer.rs

```diff
@@ -305,11 +319,35 @@ impl SchedulerInner {
             }
             let Reverse(TimeOrdered(timed_out)) = self.heap.pop().expect("It always have an item");
 
+            let timer_name = timed_out.timer_name;
+            let schedule_id = timed_out.schedule_id;
+
             if timed_out.repeat.is_some() {
                 if let Some(callback) = Callback::from_schedule(&timed_out) {
                     // timed_out.state_control is re-pushed only after it is popped (Rule 3)
-                    self.try_reschedule(timed_out.clone());
-                    worker_queue.enqueue(callback);
+                    if self.try_reschedule(timed_out.clone()) {
+                        ctrace!(
+                            TIMER,
+                            "handle_timeout(TimerName({}), {:?}, repeat): Enqueue to worker queue",
+                            timer_name,
+                            schedule_id
+                        );
+                        worker_queue.enqueue(callback);
```

Comment:
It is safe to do that since `try_reschedule` returns `false` only if the schedule was detached, which is canceled.
Even if we enqueue them, worker_loop will simply drop it.
Anyway, I'll separate the commit to clarifying that.